applications/pelion_client/src/modules/net_ot.c

```diff
@@ -122,9 +125,22 @@ static void send_net_state_event(enum net_state state)
 
 static void set_net_state(enum net_state state)
 {
-	if (state != net_state) {
-		net_state = state;
-		send_net_state_event(net_state);
+	if (state == net_state) {
+		return;
+	}
+
+	net_state = state;
+
+	if (net_state == NET_STATE_CONNECTED) {
+		k_delayed_work_cancel(&connecting_work);
+#if IS_ENABLED(CONFIG_PELION_CLIENT_POWER_MANAGER_EVENTS)
```

Comment:
Can we also remove use `if(IS_ENABLED(...)) {...}` syntax here?

Reply:
For consistency between `ot` and `lte` implementationts.

Reply:
Done